apps/site/hooks/react-client/useLocalStorage.ts

```diff
@@ -0,0 +1,22 @@
+'use client';
+import { useEffect, useState } from 'react';
+
+const useLocalStorage = <T>(key: string, initialValue?: T) => {
```

Comment:
We can only access Local storage from the client components, which causes a layout shift;
<img width="400" alt="image" src="https://github.com/user-attachments/assets/80964d56-3eaa-4472-8faa-78a60719c16a">
<img width="400" alt="image" src="https://github.com/user-attachments/assets/e3a52330-4f17-4dec-b6c0-491bc5323c88">

AFAIK we can access [cookies](https://nextjs.org/docs/app/api-reference/functions/cookies) from the server components, which might be a better approach for this setup

Also, if there's someone we can consult before storing user preferences in local storage or cookies without their consent, it would be better to ask. To be honest, I don't know anything about this topic 🙇

Reply:
+1 for using cookies.
But as far as I know, the use of local storage is not subject to regulation, but using cookies is.

Reply:
Got it, then let's continue with local storage. Once we are sure, we can consider using cookies 👍